# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsDOMWindowUtils.cpp
# Commit: 3b7f350adae9
# Full Hash: 3b7f350adae9812f4f2a571fc070f9f8b1fb55a9
# Author: Timothy Nikkel <tnikkel@gmail.com>
# Date: 2022-02-12 09:47:43
# Regressor Bug: 1754436
# File Overlap Count: 2
# Description:
#   Bug 1754662. Refactor nsLayoutUtils::GetPopupFrameForEventCoordinates. r=emilio
#   
#   This lets us re-use code for bug 1754436.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138392
# ==============================================================================

diff -r 761755ec9388 -r 3b7f350adae9 dom/base/nsDOMWindowUtils.cpp
--- a/dom/base/nsDOMWindowUtils.cpp	Sat Feb 12 05:52:14 2022 +0000
+++ b/dom/base/nsDOMWindowUtils.cpp	Sat Feb 12 08:46:47 2022 +0000
@@ -2378,11 +2378,8 @@
 
   if (message == eQueryCharacterAtPoint) {
     // Looking for the widget at the point.
-    WidgetQueryContentEvent dummyEvent(true, eQueryContentState, widget);
-    dummyEvent.Init(options);
-    InitEvent(dummyEvent, &pt);
-    nsIFrame* popupFrame = nsLayoutUtils::GetPopupFrameForEventCoordinates(
-        presContext->GetRootPresContext(), &dummyEvent);
+    nsIFrame* popupFrame = nsLayoutUtils::GetPopupFrameForPoint(
+        presContext->GetRootPresContext(), widget, pt);
 
     LayoutDeviceIntRect widgetBounds = widget->GetClientBounds();
     widgetBounds.MoveTo(0, 0);